theurgy.lic

```diff
@@ -331,146 +549,138 @@ class Theurgy
-      find_research = research_type.find { |r| find_current_project =~ /#{r}/i }
-      @hold_research = find_research
-      return if @hold_research.nil?
-    when /complete with a portion/
-      research_type = %w[Fundamental Augmentation Stream Sorcery Utility Warding]
-      find_research = research_type.find { |r| find_current_project =~ /#{r}/i }
-      @hold_research = find_research
-      return if @hold_research.nil?
-      if find_current_project[1, 2].to_i >= 60
-        wait_time = (101 - find_current_project[1, 2].to_i) * 3
-        Flags.reset('research_done')
-        pause wait_time
-      else
-        echo 'stopping research'
-      end
-      fput 'prep bless'
+    percent_complete = find_current_project[1, 2].to_i
+    if percent_complete >= 60
+      wait_time = (101 - find_current_project[1, 2].to_i) * 3
+      Flags.reset('research_done')
+      pause wait_time
+    else
+      echo 'stopping research'
+      fput 'RESEARCH CANCEL'
     end
-    @hold_research
   end
 
-  def commune_check
-    if !Flags['theurgy-commune']
-      next_commune
-    elsif Flags['theurgy-commune'].include? 'fully prepared to seek assistance from the Immortals once again'
-      next_commune
+  def continue_research
+    if @research_topic && !Flags['research_done']
+      fput "RESEARCH #{@research_topic.upcase} 300"
     end
   end
+end
+
+class Theurgy
+  include DRC
 
-  def next_commune
-    if !Flags['theurgy-eluned'] && DRStats.circle > 3 && DRSkill.getrank('Outdoorsmanship') >= 20 && DRSkill.getxp('Theurgy') <= 31
-      comm_eluned
-    elsif !Flags['theurgy-tamsine'] && DRStats.circle > 2 && DRSkill.getxp('Theurgy') <= 31
-      comm_tamsine
-    elsif !Flags['theurgy-truffenyi'] && DRStats.circle > 60 && DRSkill.getxp('Theurgy') <= 24
-      comm_truff
-    elsif !Flags['theurgy-kertigen'] && DRStats.circle > 8 && DRSkill.getxp('Theurgy') <= 31
-      comm_kert
+  def pray
+    waitrt?
+    if @favor_god
+      bput("pray #{@favor_god}", @favor_god, 'Lady of healing', 'reward of hard effort', 'sign with your hand', 'god of the Void', 'bane of accursed', 'blessing be upon your love', 'honorable and true', 'glory shine on us', 'fire across the lands', 'madness and pain', 'floods strike down', 'Lady of supreme beauty')
+    else
+      bput('pray meraud', 'meraud')
     end
+    waitrt?
   end
 
   def initialize
-    settings = get_settings
-    @theurgy_supply_container = settings.theurgy_supply_container
-    @water_holder = settings.water_holder
-    @hometown = settings.hometown
-    @immortal_aspect = settings.immortal_aspect
-    @flint_lighter = settings.flint_lighter
-    @favor_god = settings.favor_god
-    @safe_room = settings.safe_room
-
-    ######################## FLAGS ########################
+    @settings = get_settings
+    @hometown = @settings.hometown
+    @town_data = get_data('town')[@hometown] || {}
+    @theurgy_data = get_data('theurgy')[@hometown] || {}
+    @merged_data = @town_data.merge(@theurgy_data)
```

Comment:
Is `theurgy_supplies` the only data used from `base-town`?

Reply:
It's also using `almsbox`, which is also used by `crossing-training`.